apps/list.c

```diff
@@ -26,9 +26,42 @@
 #include "opt.h"
 #include "names.h"
 
-static int verbose = 0;
+static int verbose;
+#ifndef OPENSSL_NO_DEPRECATED_3_0
+static int include_legacy;
+#endif
 static const char *select_name = NULL;
 
+/* Checks to see if algorithms are fetchable */
+#define IS_FETCHABLE(type, TYPE)                                \
+    static int is_ ## type ## _fetchable(const TYPE *alg)       \
+    {                                                           \
+        TYPE *impl;                                             \
+        const char *propq = app_get0_propq();                   \
+        const char *name = TYPE ## _name(alg);                  \
+                                                                \
+        ERR_set_mark();                                         \
+        impl = TYPE ## _fetch(NULL, name, propq);               \
+        ERR_pop_to_mark();                                      \
+        if (impl == NULL)                                       \
+            return 0;                                           \
+        TYPE ## _free(impl);                                    \
+        return 1;                                               \
+    }
+IS_FETCHABLE(cipher, EVP_CIPHER)
+IS_FETCHABLE(digest, EVP_MD)
+IS_FETCHABLE(mac, EVP_MAC)
+IS_FETCHABLE(kdf, EVP_KDF)
+IS_FETCHABLE(rand, EVP_RAND)
+IS_FETCHABLE(keymgmt, EVP_KEYMGMT)
+IS_FETCHABLE(signature, EVP_SIGNATURE)
+IS_FETCHABLE(kem, EVP_KEM)
+IS_FETCHABLE(asym_cipher, EVP_ASYM_CIPHER)
+IS_FETCHABLE(keyexch, EVP_KEYEXCH)
+IS_FETCHABLE(decoder, OSSL_DECODER)
+IS_FETCHABLE(encoder, OSSL_ENCODER)
+
+#ifndef OPENSSL_NO_DEPRECATED_3_0
```

Comment:
You should wrap `list_md_fn` in the same manner

Reply:
Good point. I was looking for a function `legacy_digest_fn` :(
I'll rename it at the same time.